### PR TITLE
Add ASAN CircleCI Run

### PR DESCRIPTION
**Pull request by siying**
ASAN run is powerful in finding memory leak bugs. Running it as a part of the pre-merge CI can help contributors avoid to merge some code with bugs.

Test Plan: Watch the test result.

### CODE REVIEW

**Comment by siying** [comment]
@mrambacher hmm... maybe. How should we proceed? Wait for #6659, or make #6659 rebase later?

**Comment by mrambacher** [comment]
@siying If this one is ready to go, why don't you merge it and I will update whatever is left for #6659 after this one is committed.

**Comment by siying** [comment]
> @siying If this one is ready to go, why don't you merge it and I will update whatever is left for #6659 after this one is committed.

I just merged those code changes in https://github.com/facebook/rocksdb/pull/7025

File: .circleci/config.yml [hunk 1]
```diff
@@ -69,6 +69,20 @@ jobs:
...
       - run: CC=clang-10 CXX=clang++-10 ROCKSDB_DISABLE_ALIGNED_NEW=1 USE_CLANG=1 make all -j32 # aligned new doesn't work for reason we haven't figured out
 
+  build-linux-clang10-asan:
+    machine:
+      image: ubuntu-1604:201903-01
+    resource_class: 2xlarge
+    steps:
+      - checkout # check out the code in the project directory
+      - run: pyenv global 3.5.2
+      - run: sudo sh -c 'echo "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-10 main" >> /etc/apt/sources.list'
```
Comment (pdillinger) [inline .circleci/config.yml:79]: Optional: you could probably simplify to `echo "stuff" | sudo tee -a /etc/apt/sources.list`

File: .circleci/config.yml [hunk 2]
```diff
@@ -69,6 +69,20 @@ jobs:
...
+      - run: sudo sh -c 'echo "deb-src http://apt.llvm.org/xenial/ llvm-toolchain-xenial-10 main" >> /etc/apt/sources.list'
+      - run: sudo apt-get update -y
+      - run: sudo apt-get install -y clang-10
+      - run: sudo apt-get install -y libgflags-dev
+      - run: SKIP_FORMAT_BUCK_CHECKS=1 COMPILE_WITH_ASAN=1=1 CC=clang-10 CXX=clang++-10 ROCKSDB_DISABLE_ALIGNED_NEW=1 USE_CLANG=1 PRINT_PARALLEL_OUTPUTS=1 make asan_check -j32 # aligned new doesn't work for reason we haven't figured out
```
Comment (pdillinger) [inline .circleci/config.yml:84]: Unintentional `=1=1` but I think still works

**Pull request closed without merging**